Automattic/newspack-theme
# Simplify block selectors so styles to be applied outside of the content area

### All Submissions:


### Changes proposed in this Pull Request:

This PR is a replacement for #1131, to work with the new above header campaign placement coming in https://github.com/Automattic/newspack-popups/pull/292.

It greatly simplifies the block selectors in the themes and custom font and colour code, so it can be picked up outside of the content area. It also has the added benefit of making the styles shorter overall, and make them easier to override with Custom CSS.

As part of testing I created some before and after screenshots, as well as some test content for both the post body and campaign body, all available here: https://cloudup.com/ca1itJfrJH7 (edited to add: the light pink area in the screenshots is the above-header campaign). I think it's a good starting point for additional testing, though I'd appreciate any new blocks/setting that you all can dream up, so we're not all testing the exact same thing. The main issue I see with these changes is any regressions in how the blocks are styled in the normal content area.

This PR does not address the campaign content width; everything fills the width of the browser window, and the full and wide alignments aren't applied. We may want to restrict the content width in some way (like the same as the theme's 1200px wide header) -- open to feedback on that! 

### How to test the changes in this Pull Request:

1. Set up a test post with a lot of different blocks and settings (see the below list for the blocks that have changed and potential concerns).
2. Set up a test campaign with a few different blocks and settings -- it's probably safe to limit this one to things like text, buttons, cover blocks, anything that's more typical to promotional material.  
3. Navigate to Customizer > Colors and set distinct custom colours; make them darker so they won't be flipped to grey when used for text. 
4. Navigate to Customizer > Typography, and set custom fonts; check the option to make accent text all-caps.
4. Turn off AMP; if you've set up your blocks right, there will be too much CSS for it all to load with AMP enabled. 
5. Flip between the master branch and this PR, for each of the themes, on your test page. Confirm that a) styles are being picked up correctly in your campaign, and b) styles remain the same in the post body.

### What's changed:

#### Blocks

The following blocks are affected by this update -- I've added a star next to the ones that are less important due to use/changes:

* Homepage posts block
   * With an accent header set
   * With and without text colour set
* Columns block
   * With and without borders
* Paragraph block
   * With and without background
   * With and without drop cap
* Button block
   * Using colours available from the palette
   * Using custom colours/gradients
   * Using outline style
   * Using outline style with different colours
* Pullquote block
   * With various settings: main colour (note: this is added inline and will not be picked up after the custom colours are changed), and solid background.
* Cover block
   * With link in content (to make sure it still picks up the white text colour)
* Group Block
* Quote Block
* Image Block
   * With a caption
* Gallery Block
   * With a caption
* Separator Block
* Table Block
* Audio Block*
* Video Block*
* Preformatted*
* Verse*
* Block File*
* Block Code*


#### Theme-specific Concerns

##### Newspack Default

##### Newspack Joseph
* Changed custom typography selectors for `wp-block-pullquote` -- it should inherit the header font.
* Changed the custom typography selector for `wp-block-button` -- unlike other themes, it should inherit the body font.
* Changed the selector for `wp-block-separator`; it should still apply the double-border style to the regular and wide style.

##### Newspack Katharine
* Simplified the custom colour selector for the `entry-meta` - need to make sure it's using the primary colour in the blocks and single posts
* Simplified the `wp-block-separator` selector -- need to make sure it's still using the dotted border.
* Changed selectors for the homepage posts block section title; need to make sure it's smaller, has a greater space underneath it, and is all caps. 

##### Newspack Nelson
* Changed selectors for the homepage posts block section title; need to make sure it's smaller, has a greater space underneath it, and is all caps. 
* Simplified the `wp-block-separator` selector -- need to make sure it's still bolder and darker

##### Newspack Sacha
* Simplified the custom colour selector for the `entry-meta` - need to make sure it's using the primary colour in the blocks and single posts
* In the custom typography, made the `blockquote`, `wp-block-quote` and `wp-block-pullquote` less specific; need to confirm that they're picking up the header font. 

### Other information:
